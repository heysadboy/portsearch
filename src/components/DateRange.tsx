import { FC } from "react";

interface IDateRange {
    startDate: string | undefined,
    endDate: string | undefined
}

const DateRange: FC<IDateRange> = ({ startDate, endDate }) => {
    return (
        <div className="ui basic label date-container">
            <i className="calendar outline icon"></i>
            <span>{startDate}</span>
            <i className="date-range-icon right arrow icon"></i>
            <span>{endDate}</span>
        </div>
    );
}

export default DateRange;