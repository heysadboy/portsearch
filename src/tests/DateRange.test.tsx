import TestRenderer from "react-test-renderer";
import DateRange from "../components/DateRange";

test("DataRange Component Testing.", () => {
    const component = TestRenderer.create(
        <DateRange startDate="2021-01-01" endDate="2021-01-31"/>
    );

    let tree: any = component.toJSON();
    expect(tree).toMatchSnapshot();

    let [startDateSpan, endDateSpan] = component.root.findAllByType("span");

    //Checking if date is being displayed correctly
    expect(startDateSpan.props.children).toEqual("2021-01-01");
    expect(endDateSpan.props.children).toEqual("2021-01-31");
});