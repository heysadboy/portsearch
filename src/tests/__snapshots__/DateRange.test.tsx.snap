// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRange Component Testing. 1`] = `
<div
  className="ui basic label date-container"
>
  <i
    className="calendar outline icon"
  />
  <span>
    2021-01-01
  </span>
  <i
    className="date-range-icon right arrow icon"
  />
  <span>
    2021-01-31
  </span>
</div>
`;
